scheduler/scheduler.go

```diff
@@ -73,64 +90,71 @@ func scheduleTask(interval time.Duration, initFunc func(), tickFunc func()) {
 	}()
 }
 
-func checkMemberStatus() {
-	db.ApplyMemberCredits()
-	db.UpdateMemberTiers()
-	rm = resourcemanager.NewResourceManager(mqttserver.NewMQTTServer(), db)
-	go rm.RemovedInvalidUIDs()
-
-	const memberGracePeriod = 46
-	const membershipMonth = 31
-
-	mailer := mail.NewMailer(db, mailApi, c)
+func (s *Scheduler) checkMemberSubscriptions() {
+	members := s.dataStore.GetMembers()
 
-	pendingRevokation, err := db.GetCommunication(mail.PendingRevokationMember.String())
+	for _, member := range members {
+		if member.Level == uint8(models.Credited) {
+			// do nothing to credited members
+			continue
+		}
+		if member.SubscriptionID == "none" || len(member.SubscriptionID) == 0 {
+			// we might need to figure out why they don't have a subscriptionID
+			log.Printf("no subscriptionID for: %s", member.Name)
+			continue
+		}
 
-	if err != nil {
-		log.Errorf("Unable to get communication %v. Err: %v", mail.PendingRevokationMember, err)
-		return
-	}
+		status, value, err := s.paymentProvider.GetSubscription(member.SubscriptionID)
+		if err != nil {
+			log.Error(err)
+		}
+		lastPayment, err := strconv.ParseFloat(value, 32)
+		if err != nil {
+			log.Error(err)
+		}
 
-	pastDueAccounts := db.GetPastDueAccounts()
-	for _, a := range pastDueAccounts {
-		if a.DaysSinceLastPayment > memberGracePeriod {
-			mailer.SendCommunication(mail.AccessRevokedLeadership, c.AdminEmail, a)
-			mailer.SendCommunication(mail.AccessRevokedMember, a.Email, a)
-			db.SetMemberLevel(a.MemberId, models.Inactive)
-		} else if a.DaysSinceLastPayment > membershipMonth {
-			if !mailer.IsThrottled(pendingRevokation, models.Member{ID: a.MemberId}) {
-				//TODO: [ML] Does it make sense to send this to leadership?  It might be like spam...
-				mailer.SendCommunication(mail.PendingRevokationLeadership, c.AdminEmail, a)
-				mailer.SendCommunication(mail.PendingRevokationMember, a.Email, a)
+		if status == "ACTIVE" {
```

Comment:
These seems to be paypal statuses, would it be alright if we could create `const` for these?